test/metabase/util/embed_test.clj

```diff
@@ -54,3 +58,26 @@
               (testing "should return false when the user has already accepted licensing terms"
                 (embed/show-static-embed-terms! false)
                 (is (= (embed/show-static-embed-terms) false))))))))))
+
+(deftest maybe-populate-initially-published-at-test
+  (let [now #t "2022-09-01T12:34:56Z"]
+    (doseq [model [:model/Card :model/Dashboard]]
+      (testing "should populate `initially_published_at` when a Card's enable_embedding is changed to true"
+        (t2.with-temp/with-temp [model card {:enable_embedding false}]
+          (is (= nil (:initially_published_at card)))
+          (t2/update! model (u/the-id card) {:enable_embedding true})
+          (is (not= nil (:initially_published_at (t2/select-one model :id (u/the-id card)))))))
+      (testing "should keep `initially_published_at` value when a Card's enable_embedding is changed to false"
+        (t2.with-temp/with-temp [model card {:enable_embedding true :initially_published_at now}]
+          (is (not= nil (:initially_published_at card)))
+          (t2/update! model (u/the-id card) {:enable_embedding false})
+          (is (= (t/offset-date-time now) (:initially_published_at (t2/select-one model :id (u/the-id card)))))))
```

Comment:
I was having a trouble comparing time from the model and time created with `#t "2023"`. It turns out they're different types. I found this works, but I'm not sure what's the ideal way of doing this.

Reply:
@metabase/backend-developers could anybody chime in here? I think it would be great to get some input here on this

Reply:
Yeah those strings are a little weird for OffsetDateTimes with zero offset. You just have to know that `#t "2023-01-01T01:01:01Z"` is a ZonedDateTime in the UTC time zone. To get an OffsetDateTime with zero offset (so UTC), you have to convert it `(t/offset-date-time #t "2023-01-01T01:01:01Z")`.

Reply:
This test looks good though!